### PR TITLE
feat: cd flow 

### PR DESCRIPTION
**Pull request by ishandhanani**
This is part 2 of the CICD pipeline for rubi-py. Here's a breakdown of the steps 

#39 - CI pipeline
1. Creates an automatic test during PR's to main
2. Lint's PR titles to ensure semantic versioning 

This PR - CD-flow
1. Adds a deployment script that checks the PR title to decide the upgrade
2. The script then runs `semantic-release-publish` which 
    * Updates the `CHANGELOG.md` file with a formatted release notes. **This is why the semantic versioning is important and should be followed for EVERY single PR**
    * Bumps the version according to the PR and updates the `pyproject.toml` version
    * Pushes the changes to the repo and creates a tagged branch
    * Run's `poetry build` and uploads the GitHub

TODO remaining:
1. Push to PyPi
    * I'll do this as the very last step so we don't have random updates
2. Auto create official releases
    * Will connect with @denverbaumgartner about this offline to discuss current behavior and if it can be adapted to what we talked about 

To see the current working progress, check out this fork -> https://github.com/ishandhanani/rubi-py-fork/tree/v3.0.1

### CODE REVIEW

**Comment by denverbaumgartner** [comment]
> 

this sounds awesome, great work @ishandhanani. one question i have, what happens if we open a PR, and then open another PR, and merge in the later PR before the former? will the version automatically resolve based on what is in `master`?

**Comment by ishandhanani** [comment]
Update - versioning should be correct now. The final tag I added in the toml file `version_source = "tag"` should read from the pyproject and update accordingly.

**Comment by ishandhanani** [comment]
Update: After testing extensively in https://github.com/ishandhanani/rubi-py-fork, this flow follows what we've been discussing. The only thing left is adding the PyPi deployment token which should complete the flow.

**Review by denverbaumgartner — APPROVED** [review]
let it rip

File: test.txt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+testing pr into testcicdclow
```
Comment (adamamsmith) [inline test.txt:1]: What is this?

File: .github/workflows/deployment_and_release.yml [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+          poetry install
+      - name: Prepare GH release   
+        run: |
+          cd rubi
+          git config user.name github-actions
+          git config user.email github-actions@github.com
+          poetry run semantic-release publish
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      #- name: Prod PyPi deployment
```
Comment (adamamsmith) [inline .github/workflows/deployment_and_release.yml:95]: Uncomment

**Review by adamamsmith — APPROVED** [review]

**Pull request merged**